test/units/modules/test_copy.py

```diff
@@ -11,102 +11,109 @@
-                  ),
-                 ('/dir1/',
-                  # 0 existing dirs: error (because / should always exist)
-                  None,
-                  # 1 existing dir:
-                  ('/', ['dir1']),
-                  # 2 existing dirs:
-                  ('/dir1', []),
-                  # 3 existing dirs: Same as 2 because we never get to the third
-                  ),
-                 ) + THREE_DIRS_DATA
-
-
-ONE_DIR_DATA = (('dir1',
-                 # 0 existing dirs:
-                 ('.', ['dir1']),
-                 # 1 existing dir:
-                 ('dir1', []),
-                 # 2 existing dirs: Same as 1 because we never get to the third
-                 ),
-                ('dir1/',
-                 # 0 existing dirs:
-                 ('.', ['dir1']),
-                 # 1 existing dir:
-                 ('dir1', []),
-                 # 2 existing dirs: Same as 1 because we never get to the third
-                 ),
-                ) + TWO_DIRS_DATA
+TWO_DIRS_DATA: tuple[tuple[str, tuple[str, list[str]] | None, tuple[str, list[str]], tuple[str, list[str]]], ...] = (
+    ('dir1/dir2',
+     # 0 existing dirs:
+     ('.', ['dir1', 'dir2']),
+     # 1 existing dir:
+     ('dir1', ['dir2']),
+     # 2 existing dirs:
+     ('dir1/dir2', []),
+     # 3 existing dirs: Same as 2 because we never get to the third
+     ),
+    ('dir1/dir2/',
+     # 0 existing dirs:
+     ('.', ['dir1', 'dir2']),
+     # 1 existing dir:
+     ('dir1', ['dir2']),
+     # 2 existing dirs:
+     ('dir1/dir2', []),
+     # 3 existing dirs: Same as 2 because we never get to the third
+     ),
+    ('/dir1',
+     # 0 existing dirs: error (because / should always exist)
+     None,
+     # 1 existing dir:
+     ('/', ['dir1']),
+     # 2 existing dirs:
+     ('/dir1', []),
+     # 3 existing dirs: Same as 2 because we never get to the third
+     ),
+    ('/dir1/',
+     # 0 existing dirs: error (because / should always exist)
+     None,
+     # 1 existing dir:
+     ('/', ['dir1']),
+     # 2 existing dirs:
+     ('/dir1', []),
+     # 3 existing dirs: Same as 2 because we never get to the third
+     ),
+)
+TWO_DIRS_DATA += tuple(item[:4] for item in THREE_DIRS_DATA)
+
+
+ONE_DIR_DATA: tuple[tuple[str, tuple[str, list[str]] | None, tuple[str, list[str]]], ...] = (
+    ('dir1',
+     # 0 existing dirs:
+     ('.', ['dir1']),
+     # 1 existing dir:
+     ('dir1', []),
+     # 2 existing dirs: Same as 1 because we never get to the third
+     ),
+    ('dir1/',
+     # 0 existing dirs:
+     ('.', ['dir1']),
+     # 1 existing dir:
+     ('dir1', []),
+     # 2 existing dirs: Same as 1 because we never get to the third
+     ),
+)
+ONE_DIR_DATA += tuple(item[:3] for item in TWO_DIRS_DATA)
 
 
 @pytest.mark.parametrize('directory, expected', ((d[0], d[4]) for d in THREE_DIRS_DATA))
+@pytest.mark.xfail(reason='broken test and/or code, original test missing assert', strict=False)
```

Comment:
Why `strict=False`? Is it flaky? Only flaky tests should be non-strict.

Reply:
The tests are parametrized, with some passing and some failing. Instead of marking each failing test case as xfail, I marked the entire test as strict=False. I'll leave it up to someone else to figure out what's wrong with the tests and/or code and to decide if it's worthwhile to mark xfail on individual test cases.